xtra-libp2p/tests/basic.rs

```diff
@@ -57,6 +59,37 @@ async fn hello_world() {
     assert_eq!(string, "Hello Bob!");
 }
 
+#[tokio::test]
+async fn blocked_peers_cannot_connect() {
+    let alice_hello_world_handler = HelloWorld::default().create(None).spawn_global();
+    let (alice, bob, _) = alice_and_blockable_bob(
```

Comment:
it's probably not worth making an abstraction for a once-off test (this function reads really fishy), especially when one of the parameter is a bool 😢

IMHO much better would be just doing calling `make_node` and `make_node_with_blocks` here. 

Then it actually reads easier and we're getting rid of all the weirdness.